### DISCUSSION TITLE
Should ARR Affinity be turned ON for  Azure Durable functions ?

### DISCUSSION THREAD

**Discussion by prabhakarreddy1234** [Azure/azure-functions-durable-extension#1768] [Q&A] [answered]
I am just wondering, as durable functions seems to be stateful, should ARR Affinity be turned ON in functions portal ?

**Comment by cgillum** [comment, marked as answer]
Thanks for the question. Turning ARR affinity on is not necessary. ARR affinity is only helpful for HTTP scenarios, where state is managed on specific VMs and the front-end load balancer needs hints about which VM to direct HTTP requests to.

In the case of Durable Functions, all messages are sent via queues, so HTTP, front-ends, and affinity cookies are not involved at all. Instead, the Durable Functions uses it's own internal mechanism involving blob leases and multiple control queues to ensure that all messages get routed to the correct worker VMs. You can find more information about our stateful partitioning strategy here: https://docs.microsoft.com/en-us/azure/azure-functions/durable-functions-perf-and-scale#orchestrator-scale-out
